### PR TITLE
Feature module split

### PR DESCRIPTION
**Pull request by Vignesh-772**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/YoutubeVideoView.java [hunk 1]
```diff
@@ -61,10 +60,10 @@ protected void onDestroy() {
             youTubePlayerView.release();
         }
         super.onDestroy();
-        if(CommonJsInterface.youtubePlayer != null) {
-            CommonJsInterface.youTubePlayerView.exitFullScreen();
-            CommonJsInterface.youtubePlayer.seekTo(duration);
-            CommonJsInterface.youtubePlayer.play();
-        }
+//        if(CommonJsInterface.youtubePlayer != null) {
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/YoutubeVideoView.java:63]: Remove commented code if not used

File: Frontend/android-native/mobility-app/src/main/res/drawable/ic_agency_cancelled.xml
Comment (Rohit4997) [inline Frontend/android-native/mobility-app/src/main/res/drawable/ic_agency_cancelled.xml:1]: Where is this file used?

File: Frontend/android-native/mobility-app/src/main/res/drawable/ic_agency_offline.xml
Comment (Rohit4997) [inline Frontend/android-native/mobility-app/src/main/res/drawable/ic_agency_offline.xml:1]: Where is this file used?

File: Frontend/output/A/externs.cbor
Comment (Rohit4997) [inline Frontend/output/A/externs.cbor:1]: Why this file is reqiured?

File: Frontend/output/A/index.js
Comment (Rohit4997) [inline Frontend/output/A/index.js:1]: Why this file is reqiured?

File: Frontend/output/cache-db.json
Comment (Rohit4997) [inline Frontend/output/cache-db.json:1]: Why this file is reqiured?

File: Frontend/packages.dhall [hunk 2]
```diff
@@ -167,8 +167,8 @@ in  upstream
       , "web-dom"
       , "web-events"
       ]
-    , repo = "https://github.com/belevy/purescript-presto-dom.git"
-    , version = "nammaYatri-p-0.15"
+    , repo = "https://github.com/Vignesh-772/purescript-presto-dom.git"
+    , version = "Add-font-weight"
```
Comment (Rohit4997) [inline Frontend/packages.dhall:171]: Why this version, Any specific reason?

**Review by Rohit4997 — DISMISSED** [review]

File: .DS_Store
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply .DS_Store:1]: I can't see that file in local

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Handler.purs [hunk 3]
```diff
@@ -18,18 +18,21 @@ module Screens.HomeScreen.Handler where
...
 import Screens.HomeScreen.View as HomeScreen
 import Types.App (FlowBT, GlobalState(..), ScreenType(..), HOME_SCREEN_OUTPUT(..))
+import Debug (spy)
+import Presto.Core.Types.Language.Flow (getLogFields)
 
 homeScreen ::FlowBT String HOME_SCREEN_OUTPUT
 homeScreen = do
   (GlobalState state) <- getState
   act <- lift $ lift $ runScreen $ HomeScreen.screen state.homeScreen
+  _ <- pure $ spy "homrscreen" "handler"
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Handler.purs:35]: if not required let's remove this log, also spelling is wrong

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs [hunk 4]
```diff
@@ -87,10 +87,12 @@ initData = {
       , id: ""
       }
     , cancelRideConfirmationData : { delayInSeconds : 5, timerID : "", enableTimer : true, continueEnabled : false }
+    , config : DC.config
+    , logField : empty
     },
     props: {
       rideRequestFlow : false
-    , isSearchLocation : NoView
+    , isSearchLocation : SearchLocation
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs:95]: Why do we need a view in dummy data? It should be NoView only

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 5]
```diff
@@ -96,7 +101,7 @@ screen initialState =
             _ <- pure $ printLog "storeCallBackCustomer initially" "."
             _ <- pure $ printLog "storeCallBackCustomer callbackInitiated" initialState.props.callbackInitiated
             -- push NewUser -- TODO :: Handle the functionality
-            _ <- isMockLocation push IsMockLocation
+            -- _ <- isMockLocation push IsMockLocation
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:104]: Why did we removed this check??

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 6]
```diff
@@ -1924,7 +1979,7 @@ confirmRide action count duration push state = do
         let status = if fareProductType == "OneWaySpecialZoneAPIDetails" then "CONFIRMED" else "TRIP_ASSIGNED"
         if  status == resp.status then do
             doAff do liftEffect $ push $ action response
-            _ <- pure $ firebaseLogEvent "ny_user_ride_assigned"
+            -- _ <- pure $ logEvent state.data.logField "ny_user_ride_assigned"
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:1982]: better remove than commenting these

File: Frontend/ui-customer/src/Services/Backend.purs [hunk 7]
```diff
@@ -8,33 +8,35 @@
 
   is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 
-  or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+  or FITNESS FOR A PARTIcuLAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Services/Backend.purs:11]: Why this spelling is changes in few of the files? :joy:

File: Frontend/ui-driver/src/Screens/EnterMobileNumberScreen/Controller.purs [hunk 8]
```diff
@@ -71,12 +71,12 @@ eval (PrimaryEditTextAction (PrimaryEditText.TextChanged valId newVal)) state =
   _ <- if length newVal == 10 then do
             pure $ hideKeyboardOnNavigation true 
             else pure unit    
-  let isValidMobileNumber = case (charAt 0 newVal) of 
-                                    Just a -> if a=='0' || a=='1' || a=='2' || a=='5' then false 
-                                                else if a=='3' || a=='4' then
-                                                    if newVal=="4000400040" || newVal=="3000300030" then true else false 
-                                                        else true 
-                                    Nothing -> true 
+  let isValidMobileNumber = true --case (charAt 0 newVal) of
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/EnterMobileNumberScreen/Controller.purs:74]: why is this removed? It's a mandatory item for getting approval on play store/ app store

**Review by witcher-shailesh — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/build.gradle [hunk 9]
```diff
@@ -123,37 +112,37 @@ android {
...
         }
-        ny_p_ {
+        nyDriver {
             applicationId "in.juspay.nammayatripartner"
             buildConfigField "String", "MERCHANT_TYPE", "\"DRIVER\""
             buildConfigField "String", "MERCHANT", "\"KA\""
             dimension "app"
         }
-        js_kl_ {
+        jsUser {
```
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/app/build.gradle:133]: It is done in separate PR

File: Frontend/android-native/mobility-app/src/main/AndroidManifest.xml
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/AndroidManifest.xml:1]: This is how the module splitting is works.

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OtpUtils.java
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OtpUtils.java:1]: It is deleted moved to HyperSDK

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 10]
```diff
@@ -108,32 +105,73 @@ public void onCreate() {
...
-        }
+    public static void registerCallback(CallBack notificationCallback) {
+        callBack.add(notificationCallback);
+    }
+
+    public static void deRegisterCallback(CallBack notificationCallback) {
+        callBack.remove(notificationCallback);
+    }
+
+    @SuppressLint("SetTextI18n")
```
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:116]: To Remove Warnings

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 11]
```diff
@@ -470,97 +424,92 @@ public int onStartCommand(Intent intent, int flags, int startId) {
...
-                if ((sheetArrayList.size()==1  && rideStatusListener!=null)){
-                    return;
-                }
-                for (int i = 0; i<sheetArrayList.size(); i++){
-                    if (id.equals(sheetArrayList.get(i).getSearchRequestId())){
-                        removeCard(i);
-                        break;
-                    }
+    public void removeCardById(String id) {
+        if (sheetArrayList.size() > 0) {
```
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:502]: It is initialised already it is not going to be null.

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java [hunk 12]
```diff
@@ -93,58 +96,50 @@ public Boolean driverRespondApi(String searchRequestId, int offeredPrice, boolea
...
+            mainLooper.post(() -> Toast.makeText(context.getApplicationContext(), "Request Timeout", Toast.LENGTH_SHORT).show());
             return false;
         } catch (Exception e) {
             return false;
         }
     }
 
 
-    public int calculateExpireTimer(String expireTimeTemp, String currTimeTemp){
+    public static int calculateExpireTimer(String expireTimeTemp, String currTimeTemp) {
```
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java:129]: No @Rohit4997 It is good to have helper functions as static.

File: Frontend/packages.dhall (hunk 2 above)
Comment (Vignesh-772) [reply Frontend/packages.dhall:171]: yes we added new prop.

File: Frontend/android-native/mobility-customer/src/main/java/in/juspay/mobility/customer/MobilityCustomerBridge.java [hunk 13]
```diff
@@ -0,0 +1,977 @@
...
+
+        JSONObject jsonObject = new JSONObject();
+
+        Location locationA = new Location("point A");
+        locationA.setLatitude(lat);
+        locationA.setLongitude(lng);
+
+        double minDist = 10000000000.0;
+
+        Location location = new Location("final point");
```
[the remark this answers is not in the document]
Comment (MercyQueen) [reply Frontend/android-native/mobility-customer/src/main/java/in/juspay/mobility/customer/MobilityCustomerBridge.java:396]: @soumyajit2304 can you please reply here.

[the remark this answers is not in the document]
Comment (soumyajit2304) [reply Frontend/android-native/mobility-customer/src/main/java/in/juspay/mobility/customer/MobilityCustomerBridge.java:396]: `new Location` takes a key as paramater.   
Which also denotes the name of that point.

File: Frontend/ui-common/src/Engineering/Helpers/Commons.purs [hunk 14]
```diff
@@ -145,6 +150,15 @@ flowRunner initialState flow = do
   let freeFlow = S.evalStateT (run runtime flow)
   try $ new (defaultState initialState) >>= freeFlow
 
+flowRunnerWithState :: forall a st. st -> (Flow st a) -> Aff (Either Error a)
```
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/ui-common/src/Engineering/Helpers/Commons.purs:153]: Previous it is needed now the normal flow runner itself will take state so not required

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**